codalab/common.py

```diff
@@ -190,6 +190,28 @@ class StorageURLScheme(Enum):
     GCS_STORAGE = "gs://"
 
 
+# A Dict mapping from StorageURLSchema to StorageType
+storage_url_type_dict = {
+    StorageURLScheme.DISK_STORAGE: StorageType.DISK_STORAGE,
+    StorageURLScheme.AZURE_BLOB_STORAGE: StorageType.AZURE_BLOB_STORAGE,
+    StorageURLScheme.GCS_STORAGE: StorageType.GCS_STORAGE,
+}
+
+
+def storage_url_to_type(url: str):
```

Comment:
I think we should replace this method with just `parse_linked_bundle_url(url).storage_type`

Reply:
Yeah If we can merge this together, it will be more clear. But for the function parse_linked_bundle_url, it seems that it only parses: an archive file on Blob Storage: "azfs://storageclwsdev0/bundles/uuid/contents.tar.gz". And when adding a storage using url (eg, "azfs://devstoreaccount1/bundles"), we do not know the uuid and file name (We only know after uploading some files). Do you think we should modify `parse_linked_bundle_url()` to parse  or still use `storage_url_to_type()` function?

Reply:
I think we should modify `parse_linked_bundle_url` to parse other storage type urls as well -- because eventually we'll need to do that anyway to support these other bundle stores. If it's just a URL "azfs://devstoreaccount1/bundles", the scheme parsing should still work, so it should be OK. We might want to add that fact to the parse_linked_bundle_url documentation and update the parse_linked_bundle_url unit tests, though.

Reply:
Thanks Ashwin! Now it will failed because of  this line `storage_account, container, bundle_uuid, contents_file, *remainder = url.split("/", 4)`. Because for  url like this "azfs://devstoreaccount1/bundles", we do not have enough parameter to parse. I will modify `parse_linked_bundle_url` to support parsing short url and modify unitests